Fix invalid logger names in tests

We're bringing in a fix to the way custom_log configuration is applied, which has highlighted a few errors that need fixing.

Some custom_logs in `test_beddays` are invalid. They should refer to the actual loggers being use in BedDays (i.e. tlo.methods.healthsystem). We have a rule that all loggers in src/tlo/methods should begin with `tlo.methods.`.

Failures in test_beddays:

```
FAILED tests/test_beddays.py::test_bed_days_basics[83563095832589325021] - AssertionError: Only logging of tlo modules is allowed; name is BedDays
FAILED tests/test_beddays.py::test_bed_days_property_is_inpatient[83563095832589325021] - AssertionError: Only logging of tlo modules is allowed; name is BedDays
FAILED tests/test_beddays.py::test_bed_days_released_on_death[83563095832589325021] - AssertionError: Only logging of tlo modules is allowed; name is BedDays
```

Failures in test_diarrhoea:

```
_ test_basic_run_of_diarrhoea_module_with_default_params[83563095832589325021] _
tests/test_diarrhoea.py:79: in test_basic_run_of_diarrhoea_module_with_default_params
    assert name.startswith('tlo'), f'Only logging of tlo modules is allowed; name is {name}'
E   AssertionError: Only logging of tlo modules is allowed; name is Diarrhoea
```

These will need to be fixed before we can fix custom_logs.

--- a/tests/test_beddays.py
+++ b/tests/test_beddays.py
@@ -139,7 +139,7 @@ def test_bed_days_basics(tmpdir, seed):
         'filename': 'bed_days',
         'directory': tmpdir,
         'custom_levels': {
-            "BedDays": logging.INFO}
+            'tlo.methods.healthsystem': logging.INFO}
     })
     sim.register(
         demography.Demography(resourcefilepath=resourcefilepath),
@@ -338,7 +338,7 @@ def test_bed_days_property_is_inpatient(tmpdir, seed):
         'filename': 'temp',
         'directory': tmpdir,
         'custom_levels': {
-            "BedDays": logging.INFO,
+            'tlo.methods.healthsystem': logging.INFO,
         }
     })
     sim.register(
@@ -462,7 +462,7 @@ def test_bed_days_released_on_death(tmpdir, seed):
         'filename': 'temp',
         'directory': tmpdir,
         'custom_levels': {
-            "BedDays": logging.INFO,
+            'tlo.methods.healthsystem': logging.INFO,
         }
     })
     sim.register(

--- a/tests/test_diarrhoea.py
+++ b/tests/test_diarrhoea.py
@@ -70,7 +70,7 @@ def test_basic_run_of_diarrhoea_module_with_default_params(tmpdir, seed):
     log_config = {'filename': 'tmpfile',
                   'directory': tmpdir,
                   'custom_levels': {
-                      "Diarrhoea": logging.INFO}
+                      "tlo.methods.diarrhoea": logging.INFO}
                   }
 
     sim = Simulation(start_date=start_date, seed=seed, log_config=log_config)

